lib/tasks_private/vpc.rake

```diff
@@ -0,0 +1,168 @@
+namespace :spec do
+  require 'ibm_cloud_sdk_core'
+  require 'ibm_cloud_resource_controller'
+  require 'ibm_vpc'
+
+  base_dir = File.join(ManageIQ::Providers::IbmCloud::Engine.config.root.to_s, 'spec')
+  cass_dir = File.join(base_dir, 'vcr_cassettes/manageiq/providers/ibm_cloud')
+  spec_dir = File.join(base_dir, 'models/manageiq/providers/ibm_cloud')
+
+  # Prerequisites:
+  # 1. Log in with IBM Cloud CLI i.e. `ibmcloud login -sso`
+  # 2. Place IBM Cloud API key in secrets config
+  desc "Generate refresher VCR"
+  task :refresh_init => :environment do
+    Dir["#{cass_dir}/vpc/cloud_manager/refresher*.yml"].each do |file|
+      FileUtils.rm(file, :verbose => true)
+    end
+    Dir["#{cass_dir}/vpc/cloud_manager/refresher/vm_target.yml"].each do |file|
+      FileUtils.rm(file, :verbose => true)
+    end
+
+    apikey = YAML.load_file("config/secrets.yml")["test"]["ibm_cloud_vpc"]["api_key"]
+    token = IBMCloudSdkCore::IamAuthenticator.new(:apikey => apikey)
+                                             .instance_variable_get(:@token_manager).access_token
+    auth = IBMCloudSdkCore::BearerTokenAuthenticator.new(:bearer_token => token)
+
+    connection = IbmVpc::VpcV1.new({:authenticator => auth})
+    resource_controller = IbmCloudResourceController::ResourceControllerV2.new({:authenticator => auth})
+
+    # Create resources
+    begin
+      default_resource_group_id = `ibmcloud resource group default --id | tr -d '\n'`
+      cloud_db_id = resource_controller.create_resource_instance(
+        :name             => 'rake-db',
+        :target           => 'bluemix-us-south',
+        :resource_group   => default_resource_group_id,
+        :resource_plan_id => 'databases-for-postgresql-standard'
+      ).result['id']
+      status = resource_controller.get_resource_instance(:id => cloud_db_id).result['state'] until status == 'active'
+    rescue => err
+      puts err
+    end
+
+    network_id = connection.create_vpc(:name => 'rake-network').result['id']
+    subnet_prototype = {
+      :vpc             => {
+        :id => network_id
+      },
+      :name            => 'rake-subnet',
+      :zone            => {
+        :name => 'us-south-1'
+      },
+      :ipv4_cidr_block => '10.240.0.0/24'
+    }
+    subnet_id = connection.create_subnet(:subnet_prototype => subnet_prototype).result['id']
+    volume_prototype = {
+      :profile  => {
+        :name => '5iops-tier'
+      },
+      :zone     => {
+        :name => 'us-south-1'
+      },
+      :name     => 'rake-vol',
+      :capacity => 10
+    }
+    volume_id = connection.create_volume(:volume_prototype => volume_prototype).result['id']
+
+    `ssh-keygen -f temp -t rsa -P ""`
+    ssh_key = `cat temp.pub`
```

Comment:
Backticks are convenient for running complex commands but this is easy enough to do in ruby with `File.read("temp.pub")`